src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.html

```diff
@@ -1,35 +1,119 @@
-            </div>
-          </th>
-        </tr>
-        <tr *ngFor="let key of leaderboard" [class.highlight]="key.is_highlighted" class="pointer" [routerLink]="routerPublic.url.split('/').length == 6 ? ['../' + key.submission__participant_team__team_name] : [key.submission__participant_team__team_name]">
-          <td><div>{{initial_ranking[key.submission__participant_team__team_name]}}</div></td>
-          <td><div class="fw-regular">{{key.submission__participant_team__team_name}}</div></td>
-          <td *ngFor="let score of key.result"><div>{{score | number : '1.2-2'}}</div></td>
-          <td><div>{{ key.submission__submitted_at_formatted }}</div></td>
-        </tr>
-    </table>
+<div class="challenge-card">
+  <div class="ev-card-panel card-bt-margin">
+    <div class="ev-md-container ev-panel-title" *ngIf="challenge.leaderboard_description">
+      <div class="row row-lr-margin">
+        <div class="col-sm-12 col-xs-12 col-lr-pad">
+          <span class="fw-regular">Description</span>
+        </div>
+        <div class="col-sm-12 md-body-1 col-lr-pad" [innerHTML]="challenge.leaderboard_description"></div>
+      </div>
+    </div>
+    <div class="ev-md-container bottom-hr-line">
+      <div (click)="refreshLeaderboard()" class="pointer update-page" *ngIf="showLeaderboardUpdate">
+        <span class="text-white">Page is Outdated, Click to update&nbsp; <i class="fa fa-refresh text-highlight"></i></span>
+      </div>
+      <div class="row row-lr-margin phase-title">
+        <div class="col-sm-12">
+          <strong class="fw-semibold content fs-15">Please select from following phases!</strong>
+        </div>
+      </div>
+      <div class="row row-lr-margin">
+        <div class="col-sm-6 col-xs-12 col-lr-pad phase-select-box">
+          <app-selectphase [phases]="filteredPhaseSplits" [phaseSelectionType]="phaseSelectionType" 
+            [phaseSelectionListType]="phaseSelectionListType" [phaseSplitSelected]="phaseSplitSelected()" 
+            #phasesplitselect>
+          </app-selectphase>
+        </div>
+      </div>
+    </div>
+
+    <div class="ev-card-body exist-team-card">
+      <div class="row row-lr-margin">
+        <div class="horizontal-scroll">
+          <div class="col-sm-12 col-lr-pad">
+            <div *ngIf="leaderboard.length <= 0 && selectedPhaseSplit == null" class="fw-regular result-wrn">No phase selected.</div>
+              <div *ngIf="leaderboard.length > 0 && selectedPhaseSplit" class="result-wrn content">
+                <mat-chip-list>
+                  <mat-chip>B</mat-chip> - Baseline submission
+                </mat-chip-list>
+              </div>
+
+              <table *ngIf="leaderboard.length > 0 && selectedPhaseSplit" class="centered highlight">
+                <thead>
+                  <tr class="content">
+                    <td data-field="rank" class="align-center">
+                      <a (click)="sortColumn!='rank' && reverseSort = false;sortColumn == 'rank' && 
```

Comment:
Push this logic to `.ts` and create a function to take care of it. (Similarly at other places in this file.)

Reply:
@lunayach done